### PR TITLE
[vividus-plugin-azure-storage-account] Adjust step which saves blob to variable to work with binary data

### PR DESCRIPTION
**Pull request by VolhaHurynovich**

### CODE REVIEW

**Review by ikalinin1 — APPROVED** [review]

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java [hunk 1]
```diff
@@ -114,9 +116,17 @@ public void downloadBlob(String blobName, String containerName, String storageAc
     {
         try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
         {
-            blobServiceClientFactory.createBlobClient(blobName, containerName, storageAccountKey)
-                    .downloadStream(outputStream);
-            variableContext.putVariable(scopes, variableName, outputStream.toString(StandardCharsets.UTF_8));
+            var blobClient = blobServiceClientFactory.createBlobClient(blobName, containerName, storageAccountKey);
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java:119]: we don't use `var` in production code yet, as we don't have clear strategy for it, it's allowed in test code only

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java [hunk 2]
```diff
@@ -114,9 +116,17 @@ public void downloadBlob(String blobName, String containerName, String storageAc
...
+            blobClient.downloadStream(outputStream);
+            if (MediaTypeRegistry.getDefaultRegistry().isInstanceOf(blobClient.getProperties().getContentType(),
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java:121]: is it possible to get blob data and its properties using a single backend call?

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java [hunk 3]
```diff
@@ -114,9 +116,17 @@ public void downloadBlob(String blobName, String containerName, String storageAc
...
+                    MediaType.TEXT_PLAIN))
+            {
+                variableContext.putVariable(scopes, variableName, outputStream.toString(StandardCharsets.UTF_8));
+            }
+            else
+            {
+                variableContext.putVariable(scopes, variableName, outputStream.toByteArray());
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java:128]: refactor to minimize copied logic

File: vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java [hunk 4]
```diff
@@ -107,25 +108,28 @@ void shouldRetrieveBlobServiceProperties()
         verify(variableContext).putVariable(SCOPES, VARIABLE, propertiesAsJson);
     }
 
-    @Test
-    void shouldDownloadBlob()
+    @ParameterizedTest
+    @CsvSource({
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java:112]: use `@ValueSource`

File: vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java [hunk 5]
```diff
@@ -107,25 +108,28 @@ void shouldRetrieveBlobServiceProperties()
...
-                return true;
-            }));
-            blobStorageSteps.downloadBlob(BLOB, CONTAINER, KEY, SCOPES, VARIABLE);
-            verify(variableContext).putVariable(SCOPES, VARIABLE, DATA);
-        });
+        downloadTextOrBinaryBlob(mediaType, DATA);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java:124]: use `@ValueSource`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java (hunk 1 above)
Comment (VolhaHurynovich) [reply vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java]: fixed

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java (hunk 3 above)
Comment (VolhaHurynovich) [reply vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java]: done

File: vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java (hunk 4 above)
Comment (VolhaHurynovich) [reply vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java]: done

File: vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java (hunk 5 above)
Comment (VolhaHurynovich) [reply vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java]: done

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java [hunk 6]
```diff
@@ -114,9 +118,12 @@ public void downloadBlob(String blobName, String containerName, String storageAc
     {
         try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
         {
-            blobServiceClientFactory.createBlobClient(blobName, containerName, storageAccountKey)
-                    .downloadStream(outputStream);
-            variableContext.putVariable(scopes, variableName, outputStream.toString(StandardCharsets.UTF_8));
+            String contentType = blobServiceClientFactory.createBlobClient(blobName, containerName, storageAccountKey)
+                    .downloadStreamWithResponse(outputStream, null, null, null, false, null, Context.NONE).getHeaders()
+                    .getValue(HttpHeaders.CONTENT_TYPE);
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java:123]: please extract
```java
blobServiceClientFactory.createBlobClient(blobName, containerName, storageAccountKey)
                    .downloadStreamWithResponse(outputStream, null, null, null, false, null, Context.NONE)
```
to a new variable

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java [hunk 7]
```diff
@@ -299,4 +305,28 @@ private void runWithBlobClient(FailableConsumer<BlobClient, IOException> testToR
...
+                }
+                catch (IOException e)
+                {
+                    throw new UncheckedIOException(e);
+                }
+                return true;
+            }), eq(null), eq(null), eq(null), eq(false), eq(null), eq(Context.NONE))).thenReturn(blobDownloadResponse);
+            HttpHeaders headers = mock(HttpHeaders.class);
+            when(blobDownloadResponse.getHeaders()).thenReturn(headers);
+            when(headers.getValue(eq("Content-Type"))).thenReturn(mediaType);
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java:327]:
```suggestion
            when(headers.getValue("Content-Type")).thenReturn(mediaType);
```

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java (hunk 6 above)
Comment (VolhaHurynovich) [reply vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java]: done

File: vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java (hunk 7 above)
Comment (VolhaHurynovich) [reply vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java]: done

File: vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java [hunk 8]
```diff
@@ -114,9 +119,13 @@ public void downloadBlob(String blobName, String containerName, String storageAc
     {
         try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
         {
-            blobServiceClientFactory.createBlobClient(blobName, containerName, storageAccountKey)
-                    .downloadStream(outputStream);
-            variableContext.putVariable(scopes, variableName, outputStream.toString(StandardCharsets.UTF_8));
+            BlobDownloadResponse blobResponse = blobServiceClientFactory.createBlobClient(blobName, containerName,
```
Comment (valfirst) [inline vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java:122]:
```suggestion
            BlobDownloadResponse blobDownloadResponse = blobServiceClientFactory.createBlobClient(blobName, containerName,
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**